### PR TITLE
Cleanup dependencies, reduce build size by ~50%

### PR DESCRIPTION
**Pull request by andylizi**
Before vs After: 6.19 MB -> 3.63 MB

* We don't need to shadow in `lombok` itself. Uses `annotationProcessor` instead. Gradle must be upgraded to at least 4.6 to support this feature, which is a version that still works on Java 7.
* `args4j-maven-plugin` is intended to use as a maven plugin. Doesn't make sense to shadow it in. Uses `args4j` directly.
* The requirement for `jaxb-api` has been removed in 81cba7d58c0ce8a334204700dca669f2ad8e16ff
* We are only using `simple-xml`, don't need the whole `retrofit`+`okhttp`+`okio` chain.
* Resolve deprecation warnings in JNA code.

All changes to Gradle Wrapper files are made by Gradle automatically (`gradle wrapper --gradle-version 4.6`).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,21 +19,25 @@ compileJava {
...
-	compile 'com.squareup.retrofit2:converter-simplexml:2.3.0'
+	compileOnly 'org.projectlombok:lombok:1.18.12'
+	annotationProcessor 'org.projectlombok:lombok:1.18.12'
 
-	implementation 'com.formdev:flatlaf:0.30'
+	compile 'args4j:args4j:2.33'
+	compile 'net.lingala.zip4j:zip4j:1.3.3'
+	compile 'net.java.dev.jna:jna-platform:5.0.0'
+	compile 'org.simpleframework:simple-xml:2.7.1'
+  implementation 'com.formdev:flatlaf:0.30'
```
Comment (MCOfficer) [inline build.gradle:40]: space indentation

Comment (andylizi) [reply build.gradle]: Wait what...? That isn't my commit at all. Weird. Reverted now.

Comment (luguina) [reply build.gradle]: @andylizi, I think that happened when your PR was merged with master (flatlaf:0.30 is required by dark mode theme).

Comment (andylizi) [reply build.gradle]: That's also what I thought, but why would Github consider "pushing to `laurent-clouet:master`" had anything to do with the `Allow edits from maintainers` feature is beyond me. It's not like he was pushing to my branch. And this had happened with previous PRs too, IIRC.

Anyway, I disabled the aforementioned feature and it shouldn't happen again.

**Comment by laurent-clouet** [comment]
Thanks for the cleanup!

**Pull request merged**